### PR TITLE
chore: handle uncaught error during audit connection params

### LINKED ISSUE #3570: Uncaught (in promise) Error: The provider does not support audit for connection parameters
### Bug description

This error is thrown on windows when I switch between rootful vs. rootless with a button on a podman machine creation page. 
uncaught-error-rootfull-switch

### Operating system

Windows 10

### Installation Method

Installer from website/GitHub releases

### Version

1.3.0

### Steps to reproduce

Create new Podman machine form - switch button for rootfull vs. rootless, check dev console.

### Relevant log output

```shell
Uncaught (in promise) Error: The provider does not support audit for connection parameters
await (async)
w @ PreferencesRenderingItemFormat.svelte:67
p @ PreferencesRenderingItemFormat.svelte:98
F @ PreferencesRenderingItemFormat.svelte:260
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by vzhukovs**
### What does this PR do?

Handle uncaught error during audit connection parameters.

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

#3570 

### How to test this PR?

Open dev console and try to switch or adjust controls on the create connection page. Dev console should contains warning message instead of uncaught error.

### CODE REVIEW

File: packages/renderer/src/lib/preferences/PreferencesConnectionCreationRendering.svelte [hunk 1]
```diff
@@ -194,7 +194,11 @@ async function handleValidComponent() {
     data[key] = value;
   }
 
-  connectionAuditResult = await window.auditConnectionParameters(providerInfo.internalId, data as AuditRequestItems);
+  try {
+    connectionAuditResult = await window.auditConnectionParameters(providerInfo.internalId, data as AuditRequestItems);
+  } catch (e: any) {
```
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesConnectionCreationRendering.svelte]: I don't think we should use any here as message might not be defined, it's unknown type by default

in some code we had to fix it as we had undefined or null then e.message was throwing an error

```typescript
} catch (err: unknown) {
  if (err instanceof Error) {
  ...
  }
}
```

Comment (vzhukovs) [reply packages/renderer/src/lib/preferences/PreferencesConnectionCreationRendering.svelte]: Thanks for pointing this, updated the PR.

**Review by benoitf — APPROVED** [review]
you might add else condition with logging `String(err)`

File: packages/renderer/src/lib/preferences/PreferencesConnectionCreationRendering.svelte [hunk 2]
```diff
@@ -194,7 +194,13 @@ async function handleValidComponent() {
     data[key] = value;
   }
 
-  connectionAuditResult = await window.auditConnectionParameters(providerInfo.internalId, data as AuditRequestItems);
+  try {
+    connectionAuditResult = await window.auditConnectionParameters(providerInfo.internalId, data as AuditRequestItems);
+  } catch (err: unknown) {
+    if (err instanceof Error) {
```
Comment (jeffmaury) [inline packages/renderer/src/lib/preferences/PreferencesConnectionCreationRendering.svelte:203]: Why do we have this special case (compared to line 90) ?

Comment (benoitf) [reply packages/renderer/src/lib/preferences/PreferencesConnectionCreationRendering.svelte:203]: if you want to use `.message` (on an Error) you need to check if the object is an instance of an Error first

if you don't want to display the `message` attribute then you can just let the toString() being called

Comment (jeffmaury) [reply packages/renderer/src/lib/preferences/PreferencesConnectionCreationRendering.svelte:203]: I don't get why we need a special case for Error and not in other parts of the codebase

**Pull request merged**